### PR TITLE
chore: add static site alias field

### PR DESCRIPTION
**Pull request by dannyrandall**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/deploy/static_site.go [hunk 1]
```diff
@@ -143,3 +148,20 @@ func (d *staticSiteDeployer) stackConfiguration(in *StackRuntimeConfiguration) (
...
 	return conf, nil
 }
+
+func (d *staticSiteDeployer) validateAlias() error {
+	if d.staticSiteMft.Alias == "" {
+		return nil
+	}
+
+	hasImportedCerts := len(d.envConfig.HTTPConfig.Public.Certificates) != 0
+	if hasImportedCerts {
```
Comment (Lou1415926) [inline internal/pkg/cli/deploy/static_site.go:158]: I wonder if people can still technically use an alias under `*.app.domain.com`, that has nothing to do with the env 🤔

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/cli/deploy/static_site.go (hunk 1 above)
Comment (dannyrandall) [reply internal/pkg/cli/deploy/static_site.go:158]: that's a great question... maybe for now i'll leave it as is (matching NLB logic) but if we have extra time we can see if it works with the cert lambda.

File: internal/pkg/cli/deploy/svc.go [hunk 2]
```diff
@@ -131,3 +133,68 @@ func validateMinAppVersion(app, svc string, appVersionGetter versionGetter, minV
...
+	alias string
+	app   *config.Application
+	env   string
+}
+
+func (e *errInvalidAlias) Error() string {
+	return fmt.Sprintf("alias %q is not supported in hosted zones managed by Copilot", e.alias)
+}
+
+func (e *errInvalidAlias) RecommmendActions() string {
```
Comment (iamhopaul123) [inline internal/pkg/cli/deploy/svc.go:186]: awesome!

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/deploy/static_site.go [hunk 3]
```diff
@@ -143,3 +148,20 @@ func (d *staticSiteDeployer) stackConfiguration(in *StackRuntimeConfiguration) (
...
+		return fmt.Errorf("cannot specify alias when env %q imports one or more certificates", d.env.Name)
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/static_site.go:159]: :wut: so the imported certificates on ELB limits if you can use `alias` for a static site?

Maybe I'm miss understanding, but my expectation would be that users can specify both `alias` and `certs` in a `Static Site` manifest

File: internal/pkg/cli/deploy/svc.go [hunk 4]
```diff
@@ -104,7 +106,7 @@ type errAppOutOfDate struct {
 }
 
 func (e *errAppOutOfDate) Error() string {
-	return fmt.Sprintf("app version must be > %s", e.neededVersion)
+	return fmt.Sprintf("app version must be >= %s", e.neededVersion)
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/svc.go:109]: :doge:

File: internal/pkg/cli/deploy/static_site.go (hunk 3 above)
Comment (dannyrandall) [reply internal/pkg/cli/deploy/static_site.go:159]: you're 100% right that it doesn't make sense - i was just thinking to start off with the same restrictions we have for NLB until we can validate (and fix, if necessary) that the cert lambda will work with imported certificates.

also to clarify - there is no `certs` field in the Static Site manifest right now - these are env level certs.

**Pull request merged**